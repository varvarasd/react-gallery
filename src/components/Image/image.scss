.content__wrapper {
    width: 1000px;
    height: 800px;
    border: 0.1px solid #cfcfcf;
}

.image__wrapper {
    position: relative;
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 40px;
}

.image__author {
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}

.image {
    max-width: 80%;
    max-height: 80%;
    transition: opacity 400ms ease 0ms;
}

@media (max-width: 768px) {
    .content__wrapper {
        width: 100%;
        height: 100%;
    }
    .image__wrapper {
        width: 100%;
        height: 100%;
    }
    .image__author {
        text-align: center;
    }
}