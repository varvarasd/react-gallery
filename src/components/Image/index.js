import React, { Fragment } from 'react';
import './image.scss';

const Image = ({url, alt}) => {
    return <Fragment>
        <div className="image__wrapper">
            <img className="image" src={url} alt={alt}/>
        </div>
    </Fragment>;
};

export default Image;

