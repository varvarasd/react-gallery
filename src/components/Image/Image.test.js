import React from 'react';
import { render } from '@testing-library/react';
import Image from '.';

describe('Image', () => {
    it('renders', () => {
        const { container } = render(<Image />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders with given props', () => {
        const { getByAltText } = render(<Image url='some/url' alt='some alt'/>);
        expect(getByAltText('some alt')).toBeTruthy();
        expect(getByAltText('some alt').src.includes('some/url')).toBe(true);
    });
});