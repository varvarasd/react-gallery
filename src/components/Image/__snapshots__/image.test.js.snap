// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders 1`] = `
<div
  class="image__wrapper"
>
  <img
    class="image"
  />
</div>
`;
